### PR TITLE
PR-template Changes

### PR DESCRIPTION
**Pull request by p-nayak11**
## Description
I included a review checklist, some changes to the titles, and made sure that the associated issues are **resolved** which will automatically close related issues.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
 ## Description
 Describe your pull request.
 
-## Associated Issue
-Issue #???
+## Associated / Resolved Issue
+Resolves # or link to clickup card
```
Comment (kfu02) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Resolves Issue # or [ClickUp card](link-to-clickup-card)
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -13,7 +13,16 @@ Issue #???
 2. Step 2
 3. Step 3
 
-Expected result: ???
+## Key Files to Review
+Main PR-related files
```
Comment (kfu02) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
Group 1
 * File 1
 * File 2
Group 2
 * File 3
 * File 4
```
unless you want people to write paragraphs or comma-separated lists of files

another format is the one I used in #1811, up to you

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -13,7 +13,16 @@ Issue #???
...
+
+## Review Checklist
+
+- [ ] **Docstrings**: All methods and classes should have the file appropriate docstrings which follow the guidelines in the "Contributing" page of our docs.
```
Comment (kfu02) [inline .github/PULL_REQUEST_TEMPLATE.md]: can you put a link to that page here?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -13,7 +13,16 @@ Issue #???
...
+- [ ] **Remove extra print statements**: Any print statements used for debugging should be removed
+- [ ]
```
Comment (kfu02) [inline .github/PULL_REQUEST_TEMPLATE.md]:
```suggestion
- [ ] **Tag reviewers**: tag some people for review and ping them on Slack
```

**Review by kfu02 — CHANGES_REQUESTED** [review]
good changes, some thoughts

**Comment by kfu02** [comment]
also when adding docstrings per your contributing guide in #1905 , I noticed for the C++ side you can add details: https://github.com/RoboJackets/robocup-software/blob/e32a2a3ecca4bde495cb84a00c70e00fbd385e07/soccer/src/soccer/planning/trajectory.hpp#L19-L24

can you add that to those docs?

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (p-nayak11) [reply .github/PULL_REQUEST_TEMPLATE.md]: If you put an issue issue number after the # it automatically resolved it for you. See associated issue for PR #1909

**Comment by p-nayak11** [comment]
> also when adding docstrings per your contributing guide in #1905 , I noticed for the C++ side you can add details:
> 
> https://github.com/RoboJackets/robocup-software/blob/e32a2a3ecca4bde495cb84a00c70e00fbd385e07/soccer/src/soccer/planning/trajectory.hpp#L19-L24
> 
> can you add that to those docs?

I will make another quick PR since I need to add specifics about documenting subclasses.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (kfu02) [reply .github/PULL_REQUEST_TEMPLATE.md]: Gotcha

Comment (kfu02) [reply .github/PULL_REQUEST_TEMPLATE.md]: How about the link to clickup card templating though?

**Review by kfu02 — APPROVED** [review]
looks good, I still suggest making it easier to do the clickup link but that's not blocking

**Pull request merged**